### ISSUE TITLE
3D cube synthesized daataset

### ISSUE THREAD

**Issue by ipod825** [chenhsuanlin/spatial-transformer-GAN#21]
Could you share the code on how you create the 3D cube dataset? It would be very helpful to test the model without running on a large dataset. Thanks.

**Comment by chenhsuanlin** [comment]
Unfortunately I no longer have access to the code to generate the 3D cube dataset, but I've uploaded the rendered [training set](https://cmu.box.com/shared/static/ocx4s4dfld5f48g72owr5mthl8qqwnx1.npz) and [test set](https://cmu.box.com/shared/static/injhuwtsunlpoyl5ye6022od861nzpko.npz). Hope this helps!

**Comment by ipod825** [comment]
Thanks! This helps a lot. May I ask what (perhaps python?) package did you use to generate these data?

**Comment by ipod825** [comment]
The training set does not contain "real" images, only but "fg" and "bg" images.

**Comment by chenhsuanlin** [comment]
Corresponding foreground/background images were rendered from the same viewpoint and configuration, so directly compositing the two together gives you the "real" images. Fake images were generated by adding geometric perturbations on the fly (the same as in the released code).

I generated this dataset using Matlab. Basically it's just drawing colored 3D planes to form the 3D cube/room and rendering from different viewpoints, so it could also be done easily in Python.
